mesos/mesos.go

```diff
@@ -188,10 +205,12 @@ func (m *Mesos) CollectMetrics(mts []plugin.MetricType) ([]plugin.MetricType, er
 							log.Warn("Attempted to collect metric ", requested.String(), " but it returned nil!")
 							continue
 						}
+						//clone ns
+						cns := core.Namespace(append([]core.NamespaceElement{}, requested...))
```

Comment:
What is cloning the namespace accomplishing here? If this is actually needed, please add a comment on how this works.